From 6e939861ffa97c4dd3f42a7dd7a2615fefa4ebd5 Mon Sep 17 00:00:00 2001
From: Dave Stevenson <dave.stevenson@raspberrypi.com>
Date: Wed, 8 Nov 2023 12:04:20 +0000
Subject: [PATCH] defconfigs: Drop FB_UDL from all Pi defconfigs

CONFIG_FB_UBL is the fbdev driver for DisplayLink USB displays.
CONFIG_UDL is the DRM driver for DisplayLink USB displays.

Both attach themselves to the same USB device alias, so both
get loaded for any attached display. You now have 2 drivers for
the same hardware, and a race as to which one will load.

DRM emulates the framebuffer interface, so disable the fbdev
driver.

https://github.com/raspberrypi/bookworm-feedback/issues/133

Signed-off-by: Dave Stevenson <dave.stevenson@raspberrypi.com>
---
 arch/arm/configs/bcm2709_defconfig   | 1 -
 arch/arm/configs/bcm2711_defconfig   | 1 -
 arch/arm/configs/bcmrpi_defconfig    | 1 -
 arch/arm64/configs/bcm2711_defconfig | 1 -
 arch/arm64/configs/bcm2712_defconfig | 1 -
 arch/arm64/configs/bcmrpi3_defconfig | 1 -
 6 files changed, 6 deletions(-)

--- a/arch/arm/configs/bcm2709_defconfig
+++ b/arch/arm/configs/bcm2709_defconfig
@@ -965,7 +965,6 @@ CONFIG_TINYDRM_ST7735R=m
 CONFIG_DRM_GUD=m
 CONFIG_FB=y
 CONFIG_FB_BCM2708=y
-CONFIG_FB_UDL=m
 CONFIG_FB_SIMPLE=y
 CONFIG_FB_SSD1307=m
 CONFIG_FB_RPISENSE=m
--- a/arch/arm/configs/bcm2711_defconfig
+++ b/arch/arm/configs/bcm2711_defconfig
@@ -983,7 +983,6 @@ CONFIG_TINYDRM_ST7735R=m
 CONFIG_DRM_GUD=m
 CONFIG_FB=y
 CONFIG_FB_BCM2708=y
-CONFIG_FB_UDL=m
 CONFIG_FB_SIMPLE=y
 CONFIG_FB_SSD1307=m
 CONFIG_FB_RPISENSE=m
--- a/arch/arm/configs/bcmrpi_defconfig
+++ b/arch/arm/configs/bcmrpi_defconfig
@@ -959,7 +959,6 @@ CONFIG_TINYDRM_ST7735R=m
 CONFIG_DRM_GUD=m
 CONFIG_FB=y
 CONFIG_FB_BCM2708=y
-CONFIG_FB_UDL=m
 CONFIG_FB_SIMPLE=y
 CONFIG_FB_SSD1307=m
 CONFIG_FB_RPISENSE=m
--- a/arch/arm64/configs/bcm2711_defconfig
+++ b/arch/arm64/configs/bcm2711_defconfig
@@ -1020,7 +1020,6 @@ CONFIG_TINYDRM_ST7735R=m
 CONFIG_DRM_GUD=m
 CONFIG_FB=y
 CONFIG_FB_BCM2708=y
-CONFIG_FB_UDL=m
 CONFIG_FB_SIMPLE=y
 CONFIG_FB_SSD1307=m
 CONFIG_FB_RPISENSE=m
--- a/arch/arm64/configs/bcm2712_defconfig
+++ b/arch/arm64/configs/bcm2712_defconfig
@@ -1023,7 +1023,6 @@ CONFIG_TINYDRM_ST7735R=m
 CONFIG_DRM_GUD=m
 CONFIG_FB=y
 CONFIG_FB_BCM2708=y
-CONFIG_FB_UDL=m
 CONFIG_FB_SIMPLE=y
 CONFIG_FB_SSD1307=m
 CONFIG_FB_RPISENSE=m
--- a/arch/arm64/configs/bcmrpi3_defconfig
+++ b/arch/arm64/configs/bcmrpi3_defconfig
@@ -935,7 +935,6 @@ CONFIG_TINYDRM_ST7735R=m
 CONFIG_DRM_GUD=m
 CONFIG_FB=y
 CONFIG_FB_BCM2708=y
-CONFIG_FB_UDL=m
 CONFIG_FB_SIMPLE=y
 CONFIG_FB_SSD1307=m
 CONFIG_FB_RPISENSE=m
